components/common/BraveCoreThemeProvider.tsx

```diff
@@ -20,9 +20,19 @@ export default function LightDarkThemeProvider (props: React.PropsWithChildren<P
     const handleDarkModeChange = (e: MediaQueryListEvent) => {
       setIsDarkMode(e.matches)
     }
+
+    const onVisibilityChange = () => {
+      if (document.visibilityState === 'visible') {
+        setIsDarkMode(darkModeMediaMatcher.matches)
+      }
+    }
+
     darkModeMediaMatcher.addEventListener('change', handleDarkModeChange)
+    document.addEventListener('visibilitychange', onVisibilityChange)
+
     return () => {
       darkModeMediaMatcher.removeEventListener('change', handleDarkModeChange)
+      document.removeEventListener('visibilitychange', onVisibilityChange)
```

Comment:
I think technically this should've been a separate commit since it affects other webuis and we may need to uplift differently but at this point I don't see it needing uplift separate to shields.